documentation/src/docs/state-synchronizer-design.md

```diff
@@ -7,24 +7,33 @@ You may obtain a copy of the License at
 
     http://www.apache.org/licenses/LICENSE-2.0
 -->
-# StateSynchronizer Design
+# State Synchronizer Design
 
-A StateSynchronizer provides a means for data to be written and read by multiple processes, while consistency is guaranteed using optimistic checks. A rough version of this API is [checked in here.](https://github.com/pravega/pravega/blob/master/client/src/main/java/io/pravega/client/state/StateSynchronizer.java)
+A State Synchronizer provides means for data to be written and read by multiple processes, while consistency is guaranteed using optimistic checks. State Synchronizer provides an abstraction by keeping a Java object consistent across multiple machines, hence all members of a `ReaderGroup` would see the same object even as it is modified. In the state synchronizer, the updates are stored in a log that can be arbitrarily long and is consistently stored and replicated. State Synchronizer provides the foundation for implementing any arbitrary replicated state machine.
```

Comment:
How about instead of these first two blocks:
"A State Synchronizer is means to have data written and read by multiple processes, while consistency is guaranteed using optimistic checks. State Synchronizer provides the abstraction of a user defined Java object which is kept in-sync consistently across multiple machines. (all hosts would see the same object even as it is modified). 

The [State Synchronizer API](https://github.com/pravega/pravega/blob/master/client/src/main/java/io/pravega/client/state/StateSynchronizer.java) Can be used to perform updates to the state object. This can be used to implement things such as replicated state machines, schema distribution, and leader election.

State Synchronizer works by storing a consistent history of updates to the state object. The updates are stored in a Pravega Stream. Pravega ensures that every process that is performing an update on the latest version of that Object. Thus the Object is coordinated across a fleet and everyone sees the same sequence of updates on the same object."

Reply:
@tkaitchuck Accepted suggestion and content edited